src/main/java/org/mule/extension/db/internal/domain/executor/BulkUpdateExecutor.java

```diff
@@ -75,6 +76,23 @@ public Object execute(DbConnection connection, Query query, List<List<QueryParam
       queryLogger.logQuery();
 
       return preparedStatement.executeBatch();
+    } catch (BatchUpdateException batchEx) {
+      int[] updateCounts = batchEx.getUpdateCounts();
+      int successfulRecords, failedRecords, noInfoAvailable;
+      successfulRecords = 0;
+      failedRecords = 0;
+      for (int i = 0; i < updateCounts.length; i++) {
+        if (updateCounts[i] >= 0) {
+          successfulRecords++;
```

Comment:
updateCounts[i] holds the number of affected records and successfulRecords is actually holding the successful operations. Perhaps would be better to return successful ops, affected rows, failed ops and ops without information available.